goc/share/generate_sshauthkeys

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/python
+
+import sys
+import os
+import pwd, grp
+import re
+import subprocess
+import json
+
+# it's much easier to switch to curl for the authenticated lookup than
+#  to use native python
+cmd = 'curl -s --cert /etc/grid-security/hostcert.pem --key /etc/grid-security/hostkey.pem https://topology.opensciencegrid.org/oasis-managers/json?vo="*"'
```

Comment:
```suggestion
cmd = 'curl -s --cert /etc/grid-security/hostcert.pem --key /etc/grid-security/hostkey.pem "https://topology.opensciencegrid.org/oasis-managers/json?vo=*"'
```

If you are going to use a single command string, and you want to quote the `*` because it's a glob character, well, _technically_ `?` is a glob character too, so you might as well quote the whole url string.

Alternatively you can avoid the shell interpretation question by splitting the cmd args yourself:

```suggestion
cmd = 'curl -s --cert /etc/grid-security/hostcert.pem --key /etc/grid-security/hostkey.pem https://topology.opensciencegrid.org/oasis-managers/json?vo=*'.split()
```

(Unless the whole point is to have bash (with -x) display the curl cmdline)

Reply:
Also, just a note, i think `curl -s` is completely silent, even on errors.  You can use `curl -sS ...` to keep error messages but not display regular informational messages.

Reply:
I'll use -sS